medicines/web/src/hooks/index.tsx

```diff
@@ -25,3 +30,34 @@ export const useSessionStorage = (key: string, initialValue: any): any => {
   };
   return [storedValue, setValue];
 };
+
+export const useLocalStorage = (key: string, initialValue: any): any => {
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      // tslint:disable-next-line: no-console
+      console.error(error);
+      return initialValue;
+    }
+  });
+
+  const setValue = (value: any, force = false) => {
+    if (!force && !window.localStorage.getItem('allowStorage')) {
+      throw Error(
+        "We don't have the user's permission to set a value in local storage.",
+      );
+    }
+    try {
+      const valueToStore =
+        value instanceof Function ? value(storedValue) : value;
+      setStoredValue(valueToStore);
+      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+    } catch (error) {
+      // tslint:disable-next-line: no-console
+      console.error(error);
+    }
+  };
+  return [storedValue, setValue];
+};
```

Comment:
I tried combining these into one function which worked something like this:

```javascript
const useStorage => (storageBackend: any) => {
  return (key: string, initialValue: any): any => {
    …
    return [storedValue, setValue];
  }
}

export const useSessionStorage = useStorage(window.sessionStorage);
export const useLocalStorage = useStorage(window.localStorage);
```

…but something doesn't like me using `window` in this context. 🤷‍♂